assets/src/onboarding-wizard/components/navigation-context-provider.js

```diff
@@ -24,13 +24,25 @@ export const Navigation = createContext();
 export function NavigationContextProvider( { children, pages } ) {
 	const [ activePageIndex, setActivePageIndex ] = useState( 0 );
 	const [ canGoForward, setCanGoForward ] = useState( true ); // Allow immediately moving forward on first page. @todo This may need to change in 2.1.
-	const { editedOptions } = useContext( Options );
+	const { editedOptions, readerModeWasOverridden } = useContext( Options );
 
 	const { theme_support: themeSupport } = editedOptions;
 
-	const currentPage = pages[ activePageIndex ];
+	const adaptedPages = useMemo( () => {
+		if ( 'reader' === themeSupport ) {
+			return pages;
+		}
+
+		if ( readerModeWasOverridden ) {
+			setActivePageIndex( ( index ) => index - 1 );
```

Comment:
I like how you pass a callback to the state-setting function. I've never gotten into the habit, but I think I'm going to start using it more.